src/style-spec/feature_filter/convert.js

```diff
@@ -187,7 +187,16 @@ function convertInOp(property: string, values: Array<any>, negate = false) {
     }
 
     if (uniformTypes && (type === 'string' || type === 'number')) {
-        return ['match', get, values, !negate, negate];
+        // Match expressions must have unique values.
+        const seen = [];
+        const uniqueValues = values.filter(v => {
+            if (seen.indexOf(v) === -1) {
+                seen.push(v);
+                return true;
+            }
+            return false;
+        });
```

Comment:
Note that this is `O(n^2)`, so might get really slow when the filter has a lot of values (although not sure how likely we are to hit this in practice). An alternative would be to sort the values and then filter linearly, since it doesn't matter what order they are in the `match` expression: 

```js
const uniqueValues = values.sort().filter((v, i) => values[i - 1] !== v);
```

Reply:
👍

Reply:
Nit: you can write this shorter without curly brackets, `return` and on one line.